### DISCUSSION TITLE
missing jsdoc/tsdoc of built-in types (for the purpose of intellisense)

### DISCUSSION THREAD

**Discussion by ducin** [statelyai/xstate#4595] [Ideas]
Hi :)

while working on the angular port I've stumbled upon the fact that integrations (and `core` possibly also) are lacking descriptive docs that would be very useful for developers (DX). It's easiest to show what I mean, currently it looks like this:

<img width="998" alt="image" src="https://github.com/statelyai/xstate/assets/375027/4e0e7fbe-1c92-4ae5-ab80-ed75a78fcdaf">

and what I suggest adding is:

![image](https://github.com/statelyai/xstate/assets/375027/d2967923-91c8-4112-bfc5-202a567e9331)

working with libraries is way easier with some desc. I think lots of inline-docs would be re-exported by wrapper libraries (`@xstate/react` etc.). And as I take a look at, e.g. [`packages/core/src/createMachine.ts`](https://github.com/statelyai/xstate/blob/138c9e4e374ae9123eb1f286dce48f2b35c6bcd9/packages/core/src/createMachine.ts#L67) I can see pretty much no docs for parameters.

**Comment by davidkpiano** [comment]
Started this here: https://github.com/statelyai/xstate/pull/4598
